io/src/main/scala/fs2/io/tcp/Socket.scala

```diff
@@ -203,9 +201,9 @@ protected[tcp] object Socket {
       ch.read(buff, timeoutMs, TimeUnit.MILLISECONDS, (), new CompletionHandler[Integer, Unit] {
         def completed(result: Integer, attachment: Unit): Unit =  {
           val took = System.currentTimeMillis() - started
-          FR.unsafeRunAsyncEffects(F.delay(cb(Right((result, took)))))(_ => ())
+          F.unsafeRunAsync(F.delay(cb(Right((result, took)))))(_ => ())
```

Comment:
I was just tihinking that this runs the task in AG's context. Not 100% sure if that's desired. One can have same Executor for `F` and AG, so then that's ok. What do you think?